### PR TITLE
Docker: add script to build and push the image

### PR DESCRIPTION
**Pull request by danielbeardsley**
This is meant to be run during a post-merge webhook or action.

### CODE REVIEW

File: backend/build-docker-and-push [hunk 1]
```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
+
+set -eu
+
+GIT_SHORT_REF=$(git rev-parse --short HEAD)
+DOCKER_TAG="strapi:$GIT_SHORT_REF"
+# iFixit AWS ECR for strapi
+ECR_TAG="884681002735.dkr.ecr.us-east-1.amazonaws.com/$DOCKER_TAG"
+
+echo "Buidling docker image"
```
Comment (sterlinghirsh) [inline backend/build-docker-and-push:10]:
```suggestion
echo "Building docker image"
```

**Comment by sterlinghirsh** [comment]
CR :zap:

**Comment by sctice-ifixit** [comment]
CR :flags:

**Review by deltuh-vee — CHANGES_REQUESTED** [review]

**Comment by cdcline** [comment]
CR 👍

File: backend/build-docker-and-push [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+set -eu
+
+GIT_SHORT_REF=$(git rev-parse --short HEAD)
+DOCKER_TAG="strapi:$GIT_SHORT_REF"
+# iFixit AWS ECR for strapi
+ECR_TAG="884681002735.dkr.ecr.us-east-1.amazonaws.com/$DOCKER_TAG"
+
+echo "Building docker image"
+docker build -t "$DOCKER_TAG" .
```
Comment (djmetzle) [inline backend/build-docker-and-push:11]: I think we should probably use Kaniko here? To avoid docker-in-docker?

Comment (andyg0808) [reply backend/build-docker-and-push:11]: Do we currently plan to run this from Docker? If not, I hear you, but I think we should wait to solve that until it's an issue.

**Review by andyg0808 — APPROVED** [review]

File: backend/build-docker-and-push (hunk 2 above)
Comment (djmetzle) [reply backend/build-docker-and-push:11]: Oh, im not sure what context this runs in is all. Just want to make sure we avoid coupling to docker _if_ we're going to run this inside docker.

**Comment by andyg0808** [comment]
un_deploy_block :+1:
If Docker-in-Docker's an issue, I suspect either QA will find it, or it'll be obvious enough when we go to use this.

**Pull request merged**